pisi/operations/remove.py

```diff
@@ -15,64 +15,59 @@
 
 
 def remove(
-    A, ignore_dep=False, ignore_safety=False, autoremove=False, force_prompt=False
+    packages, ignore_dep=False, ignore_safety=False, autoremove=False, force_prompt=False
 ):
     """remove set A of packages from system (A is a list of package names)"""
 
     componentdb = pisi.db.componentdb.ComponentDB()
     installdb = pisi.db.installdb.InstallDB()
 
-    A = [str(x) for x in A]
+    should_ignore_safety = (not ctx.get_option("ignore_safety")
+                            and not ctx.config.values.general.ignore_safety
+                            and not ignore_safety)
+
+    packages = [str(package) for package in packages]
 
     # filter packages that are not installed
-    A_0 = A = set(A)
+    filtered = packages = set(packages)
 
-    if (
-        not ctx.get_option("ignore_safety")
-        and not ctx.config.values.general.ignore_safety
-        and not ignore_safety
-    ):
+    if should_ignore_safety:
         if componentdb.has_component("system.base"):
             systembase = set(componentdb.get_union_component("system.base").packages)
-            refused = A.intersection(systembase)
+            refused = packages.intersection(systembase)
             if refused:
                 raise pisi.Error(
                     _("Safety switch prevents the removal of " "following packages:\n")
                     + util.format_by_columns(sorted(refused))
                 )
-                A = A - systembase
```

Comment:
As far as I can tell, this was unreachable. At least, PyCharm warned  that this code was unreachable.